src/app/modules/attachment/attachment.service.ts

```diff
@@ -0,0 +1,51 @@
+import crypto from 'crypto'
+import { ResultAsync } from 'neverthrow'
+
+import { aws as AwsConfig } from '../../../config/config'
+import { createLoggerWithLabel } from '../../../config/logger'
+import { AttachmentUploadError } from '../core/core.errors'
+
+const logger = createLoggerWithLabel(module)
+
+export const uploadAttachments = (
+  formId: string,
+  attachmentData: Record<string, unknown>,
+): ResultAsync<Map<string, string>, AttachmentUploadError> => {
+  const attachmentMetadata = new Map()
```

Comment:
I think we should specify that this is a `new Map<string, string>()`, as it gets typed as `Map<any, any>` otherwise. or you could go with my other suggestion to define a type alias

Reply:
Changed to `AttachmentMetadata`

Reply:
ah never mind, typescript doesn't support a straight new AttachmentMetadata() thing. changing to Map<string, string>